nautilus_trader/model/data.pyx

```diff
@@ -648,6 +648,13 @@ cdef class BarType:
             aggregation_source
         )
 
+        self.underlying_bar_type_is_set = False
+
+    cpdef BarType set_underlying_bar_type(self, BarType bar_type):
```

Comment:
Do we need this method, or could it be handled in the constructor?

I think it would be preferable to avoid this mutable state, and keep the type immutable.

Reply:
yes I agree on this, this is from my first version of the code, but now that I know what to do on the rust side fro bar type, or at least I can change the code, I will make this go away today. I need to go back to python side in order to finish, the rust side took me a few days as I wasn't used to it.